frontend/lib/api/ApiClient.ts

```diff
@@ -12,27 +12,44 @@ export interface ApiResponse<T> {
   data: T;
 }
 
-export type ApiError = {
+export class ApiError extends Error {
```

Comment:
Nice that you use an actual `Error`! 👍 

You can use https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties to clean this up a bit

```ts
export class ApiError extends Error {
  constructor(
    public status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError,
    public code: number,
    public error?: string,
  ) {
    super(error);
  }

  withContext(message: string): this {
    this.error = message;

    return this;
  }
}
```

Reply:
Thanks :+1: I like it clean :-)